src/main/java/sirius/kernel/xml/Outcall.java

```diff
@@ -422,6 +427,14 @@ private void connect() throws IOException {
             request = requestBuilder.build();
         }
 
+        int connections = 0;
+        do {
+            makeRequest();
+            connections++;
+        } while (connections < MAX_REDIRECTS && checkAndBuildRedirectRequest(response));
+    }
```

Comment:
I'd initialize attempts = MAX_REDIRECTS and then use a while(attempts-- > 0) {..} loop, as these are simpler to reason about.

also I'd rename makeRequest to performRequest and let it return the status code. This code could then be used to check if another repetition is needed. This would illustrate the controlflow a bit:
```
while(attempts-- > 0) {
   int statusCode = performRequest();
   if (redirectMode != NEVER && redirectReceived(statusCode)) {
      adaptRequestToRedirect();
   } else {
      return;
   }
}
```